app/src/main/java/com/duckduckgo/app/global/KeyBoardVisibilityNotifier

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2020 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.global
+
+import android.content.Context
+import android.util.TypedValue
+import android.view.View
+import android.view.ViewTreeObserver
+
+
+class KeyBoardVisibilityNotifier(context: Context) {
```

Comment:
I am aware that currently there is no official method to detect that IME is shown in Android. According to some google devs, we should not checking IME visibility in their opinion, but I agree that there are some possible use cases where it could be interesting to do so.

I wonder if we should rely on this workaround (knowing that it's kind of hacky) or if we should just think about another approach to do the same that doesn't implies relying on the IME visibility.

For example, I can think about some possible scenarios where this may not work:
* Users using an external bluetooth keyboard
* How this reacts if user uses our app in split-screen mode